src/System.Linq.Expressions/tests/InterpreterTests.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            ParameterExpression x = Expression.Parameter(typeof(int), "x");
+            Expression<Func<int, int>> f =
+                Expression.Lambda<Func<int, int>>(
+                    Expression.TryCatchFinally(
+                        Expression.Call(
+                            typeof(Math).GetMethod(nameof(Math.Abs), new[] { typeof(int) }),
+                            Expression.Divide(
+                                Expression.Constant(42),
+                                x
+                            )
+                        ),
+                        Expression.Empty(),
+                        Expression.Catch(
+                            typeof(DivideByZeroException),
+                            Expression.Constant(-1)
+                        )
+                    ),
+                    x
+                );
+
+            f.VerifyInstructions(
+                @"object lambda_method(object[])
+                  {
+                    .locals 2
+                    .maxstack 3
+                    .maxcontinuation 1
+                  
+                    IP_0000: InitParameter(0)
+                    .try
+                    {
+                      IP_0001: EnterTryFinally[0] -> 11
+                      IP_0002: LoadObject(42)
+                      IP_0003: LoadLocal(0)
+                      IP_0004: Div()
+                      IP_0005: Call(Int32 Abs(Int32))
+                      IP_0006: Goto[1] -> 13
+                    }
+                    catch(DivideByZeroException) [8->11]
+                    {
+                      IP_0007: EnterExceptionHandler()
+                      IP_0008: StoreLocal(1)
+                      IP_0009: LoadObject(-1)
+                      IP_0010: LeaveExceptionHandler[3] -> 6
+                    }
+                    finally
+                    {
+                      IP_0011: EnterFinally[0] -> 11
+                      IP_0012: LeaveFinally()
+                    }
+                  }");
+        }
+
+        public static void VerifyInstructions(this LambdaExpression expression, string expected)
+        {
+            var actual = expression.GetInstructions();
+
+            var nExpected = Normalize(expected);
+            var nActual = Normalize(actual);
+
+            Assert.Equal(nExpected, nActual);
+        }
+
+        private static string Normalize(string s)
+        {
+            var lines =
+                s
+                .Replace("\r\n", "\n")
+                .Split(new[] { '\n' })
+                .Select(line => line.Trim())
+                .Where(line => line != "" && !line.StartsWith("//"));
+
+            return string.Join("\n", lines);
+        }
+
+        private static string GetInstructions(this LambdaExpression expression)
+        {
+            Delegate d = expression.Compile(true);
+            var thunk = (Func<object[], object>)d.Target;
+            var lambda = (LightLambda)thunk.Target;
```

Comment:
I think we cannot guarantee that Target will convert to LightLambda since generally preferInterpretation does not gurantee interpretation.
Inability to get LightLambda will cause a failure of the calling test one way or another, but perhaps it can be more graceful if this returns something like  "not a LightLambda"  when Target is not LightLambda.